softwareComponents/configuration/json_include/configuration/serialization.hpp

```diff
@@ -448,46 +450,50 @@ namespace rofi::configuration::serialization {
         // function for translation of components to docs if necessary
         auto f = [ &world ]( ModuleId id ) -> ModuleType { return world.getModule( id )->type; };
 
-        for ( const auto& jj : j[ "moduleJoints" ] ) {
-            roficom::Orientation o = roficom::stringToOrientation( jj[ "orientation" ] )
-                                             .get_or_throw_as< std::runtime_error >();
+        if ( j.contains( "moduleJoints" ) ) {
+            for ( const auto& jj : j[ "moduleJoints" ] ) {
+                roficom::Orientation o = roficom::stringToOrientation( jj[ "orientation" ] )
+                                                .get_or_throw_as< std::runtime_error >();
 
-            auto [ sourceModule, sourceConnector ] = details::connectionComponentFromJSON( jj[ "from" ], f );
-            auto [ destinationModule, destinationConnector ] = details::connectionComponentFromJSON( jj[ "to" ], f );
+                auto [ sourceModule, sourceConnector ] = details::connectionComponentFromJSON( jj[ "from" ], f );
+                auto [ destinationModule, destinationConnector ] = details::connectionComponentFromJSON( jj[ "to" ], f );
 
-            auto conn = connect( world.getModule( sourceModule )->connectors()[ sourceConnector ]
-                               , world.getModule( destinationModule )->connectors()[ destinationConnector ]
-                               , o );
+                auto conn = connect( world.getModule( sourceModule )->connectors()[ sourceConnector ]
+                                , world.getModule( destinationModule )->connectors()[ destinationConnector ]
+                                , o );
 
-            details::processAttributes( jj, attrCb, conn );
+                details::processAttributes( jj, attrCb, conn );
```

Comment:
Toto je docela dlouhý štrůdl "hloupých operací". Nestálo by za to z nich vyrobit funkci/pojmenovanou labmdu (kvůli kontextu) a volat ji v tom cyklu? Pak z cyklu vidím "tady prostě načítám jeden modul".

Reply:
To jsem neřešil - nechce se mi předělávat parsování z jsonu, jenom jsem přidal, že to nespadne když to tam místo prázdného array není vůbec.

Reply:
A jakoby já jsem za to, že by se to mělo celý předělat, ale dělat se mi to nechce dokud nebudu mít pokrok na diplomce a zkoušky za sebou.

Reply:
Dobře, vyrob issue ať na to nezapomeneme a můžeš mergovat.

Reply:
Přidal jsem comment do #215